vllm/config.py

```diff
@@ -175,6 +190,7 @@ def __init__(
         if not self.skip_tokenizer_init:
             self._verify_tokenizer_mode()
         self._verify_embedding_mode()
+        self._verify_simple_mode()
```

Comment:
Let's combine these two functions together and move the logic of determining model mode to model registry. (May have to move `ModelMode` as well to avoid circular imports)